### PR TITLE
Add ability to package non-release packages

### PR DESCRIPTION
**Pull request by TravisEz13**
For certain testing, we need builds with symbols.  Adding the ability to package builds with symbols for this purpose

I enabled for all packaging types except AppImage.  

I forced the option to add `symbols` into the package name, except for Release/Linux (the Linux Release configuration).  This will allow this special package type to be easily identified.

### CODE REVIEW

File: src/powershell-unix/powershell-unix.csproj [hunk 1]
```diff
@@ -45,6 +45,11 @@
     <DefineConstants>$(DefineConstants);UNIX</DefineConstants>
   </PropertyGroup>
 
+  <PropertyGroup Condition=" '$(Configuration)' == 'LinuxDebug' ">
+    <DebugType>portable</DebugType>
```
Comment (daxian-dbw) [inline src/powershell-unix/powershell-unix.csproj]: I believe .NET CLI generates portable pdb by default on Linux/OSX. See https://github.com/OmniSharp/omnisharp-vscode/wiki/Portable-PDBs#how-to-generate-portable-pdbs

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -106,6 +110,14 @@ function Start-PSPackage {
         }
     }
 
+    if($Configuration -ine 'Release' -and $NameSuffix)
+    {
+        $NameSuffix = "$Configuration-$NameSuffix"
+    }
+    elseif ($Configuration -ine 'Release') {
+        $NameSuffix = $Configuration
```
Comment (daxian-dbw) [inline tools/packaging/packaging.psm1]: Just want to double check -- we don't change the existing names of the release packages, right?

File: src/powershell-unix/powershell-unix.csproj (hunk 1 above)
Comment (adityapatwardhan) [inline src/powershell-unix/powershell-unix.csproj]: Yes, even for Windows the PDBs that are generated are portable. Hence, for code coverage we explicitly make the DebugType as full, which is Windows style full PDBs.

**Review by adityapatwardhan — DISMISSED** [review]

File: tools/packaging/packaging.psm1 (hunk 2 above)
Comment (TravisEz13) [inline tools/packaging/packaging.psm1]: Yes, but, I'll update this code, as when I wrote it I assumed `Release` was the only Release configuration

File: src/powershell-unix/powershell-unix.csproj (hunk 1 above)
Comment (TravisEz13) [inline src/powershell-unix/powershell-unix.csproj]:
```
PS /Users/travisplunk/git/powershell> dir  /Users/travisplunk/git/powershell/src/powershell-unix/bin/Linux/netcoreapp2.0/osx.10.12-x64/publish/*.pdb -Recurse          


    Directory: /Users/travisplunk/git/powershell/src/powershell-unix/bin/Linux/netcoreapp2.0/osx.10.12-x64/publish


Mode                LastWriteTime         Length Name                                                                                                                 
----                -------------         ------ ----                                                                                                                 
------           8/7/17   9:57 AM            236 Microsoft.PowerShell.SDK.pdb                                                                                         
------           8/7/17   9:57 AM            488 powershell.pdb                                                                                                       


PS /Users/travisplunk/git/powershell> dir  /Users/travisplunk/git/powershell/src/powershell-unix/bin/LinuxDebug/netcoreapp2.0/osx.10.12-x64/publish/*.pdb -Recurse     


    Directory: /Users/travisplunk/git/powershell/src/powershell-unix/bin/LinuxDebug/netcoreapp2.0/osx.10.12-x64/publish


Mode                LastWriteTime         Length Name                                                                                                                 
----                -------------         ------ ----                                                                                                                 
------           8/5/17   9:10 AM         184768 Microsoft.PowerShell.Commands.Management.pdb                                                                         
------           8/5/17   9:10 AM         242416 Microsoft.PowerShell.Commands.Utility.pdb                                                                            
------           8/5/17   9:10 AM          78996 Microsoft.PowerShell.ConsoleHost.pdb                                                                                 
------           8/5/17   9:10 AM          29848 Microsoft.PowerShell.CoreCLR.AssemblyLoadContext.pdb                                                                 
------           8/5/17   9:10 AM          39696 Microsoft.PowerShell.CoreCLR.Eventing.pdb                                                                            
------           8/5/17   9:10 AM          87268 Microsoft.PowerShell.PSReadLine.pdb                                                                                  
------           8/5/17   9:10 AM            236 Microsoft.PowerShell.SDK.pdb                                                                                         
------           8/5/17   9:10 AM          42868 Microsoft.PowerShell.Security.pdb                                                                                    
------           8/5/17  11:42 AM            488 powershell.pdb                                                                                                       
------           8/5/17   9:10 AM        2916020 System.Management.Automation.pdb          
```

Comment (TravisEz13) [inline src/powershell-unix/powershell-unix.csproj]: If they are generated, they are not included in the publish directory.

Comment (daxian-dbw) [inline src/powershell-unix/powershell-unix.csproj]: They are generated. However, when specifying `-Crossgen`, those .pdb files will be removed during `Start-Crossgen`.

**Comment by TravisEz13** [comment]
I changed this to only allow packaging with symbols for now by removing crossgen from the build.  I've updated the PR Description.

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@adityapatwardhan The comments are addressed. Can you please take another look?

**Comment by TravisEz13** [comment]
@adityapatwardhan  any update?

**Review by adityapatwardhan — APPROVED** [review]
LGTM

**Pull request merged**